### ISSUE TITLE
Saving and Loading a Dataset returns a different elements

### ISSUE THREAD

**Issue by balvisio** [tensorflow/tensorflow#56700]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

2.9.1

### Custom Code

No

### OS Platform and Distribution

MacOS Monterey

### Python version

3.7

### Current Behaviour?

I am trying to save and subsequently load a dataset. I am observing that after loading the dataset the elements of it are many times different from the saved one.

In order to run the code I have two text files in the following structure with contents:
```shell
test
 |-neg->file.txt: Once again Mr. Costner has dragged out a movie
 |-pos->file.txt: I went and saw this movie last 
```

I am expecting that the two loops should print an identical Tensor.

### Standalone code to reproduce the issue

```python
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers

import pickle


def create_int_ds(test_ds):
    max_sequence_length = 10
    max_tokens = 20000
    text_vectorization = layers.TextVectorization(
        max_tokens=max_tokens,
        output_mode="int",
        output_sequence_length=max_sequence_length,
    )
    
    text_only_train_ds = test_ds.map(lambda x, y: x)
    text_vectorization.adapt(text_only_train_ds)
    int_test_ds = test_ds.map(
        lambda x, y: (text_vectorization(x), y), num_parallel_calls=4)
    return int_test_ds


batch_size = 1
test_ds = keras.utils.text_dataset_from_directory("test", batch_size=batch_size)
int_test_ds = create_int_ds(test_ds)

take_test_ds = int_test_ds.take(1)
print(f"type of dataset: {type(take_test_ds)}")
print(f"spec: {take_test_ds.element_spec}")

for e in take_test_ds:
    print(e)

# Save
tf.data.experimental.save(take_test_ds, "int_test_ds")
with open('element_spec', 'wb') as out_:  # also save the element_spec to disk for future loading
    pickle.dump(take_test_ds.element_spec, out_)

# Load
with open('element_spec', 'rb') as in_:
    es = pickle.load(in_)
new_take_test_ds = tf.data.experimental.load("int_test_ds", element_spec=es)

print(type(new_take_test_ds)) 
for e in new_take_test_ds:
    print(e) # Shoud print the same as the previous for loop
```

### Relevant log output

```
type of dataset: <class 'tensorflow.python.data.ops.dataset_ops.TakeDataset'>
spec: (TensorSpec(shape=(None, 10), dtype=tf.int64, name=None), TensorSpec(shape=(None,), dtype=tf.int32, name=None))
(<tf.Tensor: shape=(1, 10), dtype=int64, numpy=array([[105, 183, 111,  16, 138, 159,  41,  10,   8, 147]])>, <tf.Tensor: shape=(1,), dtype=int32, numpy=array([0], dtype=int32)>)
<class 'tensorflow.python.data.experimental.ops.io._LoadDataset'>
(<tf.Tensor: shape=(1, 10), dtype=int64, numpy=array([[  4,  64,   7,  39,  11,   8, 121, 108, 184, 175]])>, <tf.Tensor: shape=(1,), dtype=int32, numpy=array([1], dtype=int32)>)
```

**Comment by balvisio** [comment]
There is no bug. `text_dataset_from_directory` shuffles the samples by default and that is why I would see different samples.
